src/lib/components/Footer/FooterLink.tsx

```diff
@@ -1,19 +1,21 @@
 import classNames from 'classnames';
-import type { ComponentProps, FC, PropsWithChildren } from 'react';
+import { forwardRef, type ComponentProps, type PropsWithChildren } from 'react';
 import { useTheme } from '../Flowbite/ThemeContext';
 
 export interface FooterLinkProps extends PropsWithChildren<ComponentProps<'a'>> {
   href: string;
 }
 
-export const FooterLink: FC<FooterLinkProps> = ({ children, className, href }) => {
+export const FooterLink = forwardRef<HTMLAnchorElement, FooterLinkProps>(({ children, className, href }, ref) => {
   const theme = useTheme().theme.footer.groupLink.link;
 
   return (
     <li className={classNames(theme.base, className)}>
-      <a href={href} className={theme.href}>
+      <a href={href} className={theme.href} ref={ref}>
         {children}
```

Comment:
BTW, I'm fine to have the `ref=` here. But I'm not sure about the remaining Next.js specific stuff.